### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by andrewstepanets**
- Implement “view movie details” functionality;
- Used useState and useEffect hooks;
- Use useCallback hook;
- Wrote custom hooks.

### CODE REVIEW

File: src/styles/constants.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  sizeXL: 880,
+  sizeXXL: 970,
+};
+
+export const mediaQueries = (key: keyof typeof breakpoints) => {
+  return (style: TemplateStringsArray | String) =>
+    `@media screen and (max-width: ${breakpoints[key]}px) { ${style} }`;
+};
+
+// colors
```
Comment (Edward-Fedoruk) [inline src/styles/constants.ts:15]: no need for this variable. It is considered to be code small when you write comment that replicate variable name

File: src/styles/constants.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// @media
```
Comment (Edward-Fedoruk) [inline src/styles/constants.ts:1]: the same comment as below

File: src/components/movies-list/posters/poster-item/poster-item.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  const handleMoreDetails = useCallback((id) => {
+    fetchMovieDetails(id);
+  }, []);
+
+  useEffect(() => {
+    setMovieDetails(movieDetails);
+  }, [movieDetails]);
+
+  const addDefaultSrc = useCallback(({ target }) => {
```
Comment (Edward-Fedoruk) [inline src/components/movies-list/posters/poster-item/poster-item.tsx:54]: what is the point of using useCallback here? I do not see any benefit

File: src/components/movie-details/movie-details.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    runtime: number;
+    genres: string[];
+    id: number;
+  };
+}
+
+export const MovieDetails: FC<MovieDetailsProps> = ({ movieDetails }) => {
+  console.log('componentMovieDetails: ', movieDetails);
+
+  const addDefaultSrc = useCallback(({ target }) => {
```
Comment (Edward-Fedoruk) [inline src/components/movie-details/movie-details.tsx:37]: you have the same code in two components. Consider to extract it

File: src/styles/constants.ts (hunk 1 above)
Comment (andrewstepanets) [reply src/styles/constants.ts:15]: Gotcha, I'll remove it

File: src/styles/constants.ts (hunk 2 above)
Comment (andrewstepanets) [reply src/styles/constants.ts:1]: Yep, good catch

File: src/components/movies-list/posters/poster-item/poster-item.tsx (hunk 3 above)
Comment (andrewstepanets) [reply src/components/movies-list/posters/poster-item/poster-item.tsx:54]: sounds good, this is just a training of using the hook

File: src/components/movie-details/movie-details.tsx (hunk 4 above)
Comment (andrewstepanets) [reply src/components/movie-details/movie-details.tsx:37]: Sure, it's make sense to combine in the next HW

**Pull request merged**